x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/completion/HuggingFaceChatCompletionModel.java

```diff
@@ -90,6 +97,6 @@ public ExecutableAction accept(HuggingFaceActionVisitor creator) {
 
     @Override
     public Integer getTokenLimit() {
-        return getServiceSettings().maxInputTokens();
+        throw new UnsupportedOperationException("Token Limit for chat completion is sent in request and not retrieved from the model");
```

Comment:
Oh I see the issue here. This is a nit (so you don't have to change it but if you want to), I think we could create a new `AbstractHuggingFaceEmbeddingModel` or something like that and have it add an abstract method that creates the requirement for `getTokenLimit`. Then have `HuggingFaceEmbeddingModel` and `HuggingFaceElserModel` extend that. Then have `HuggingFaceRequestManager` take a `AbstractHuggingFaceEmbeddingModel`. That we I believe we'll be able to remove this override here.

Reply:
> I would actually leave it as is with potential of revisiting it some time in the future during some refactoring session.

Yeah that's fine, we can do it later 👍